client/sites-dashboard-v2/controller.tsx

```diff
@@ -0,0 +1,135 @@
+import { isEnabled } from '@automattic/calypso-config';
+import {
+	DEFAULT_SITE_LAUNCH_STATUS_GROUP_VALUE,
+	siteLaunchStatusGroupValues,
+} from '@automattic/sites';
+import { Global, css } from '@emotion/react';
+import { removeQueryArgs } from '@wordpress/url';
+import AsyncLoad from 'calypso/components/async-load';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import { setAllSitesSelected } from 'calypso/state/ui/actions';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import SitesDashboardV2 from './sites-dashboard';
+import type { Context, Context as PageJSContext } from '@automattic/calypso-router';
+
+const getStatusFilterValue = ( status?: string ) => {
+	return siteLaunchStatusGroupValues.find( ( value ) => value === status );
+};
+
+function getQueryParams( context: Context ) {
+	return {
+		page: context.query.page ? parseInt( context.query.page ) : undefined,
+		perPage: context.query[ 'per-page' ] ? parseInt( context.query[ 'per-page' ] ) : undefined,
+		search: context.query.search,
+		status: context.query.status,
+	};
+}
+
+export function sanitizeQueryParameters( context: PageJSContext, next: () => void ) {
+	/**
+	 * We need a base case because `page.replace` triggers a re-render for every middleware
+	 * in the route.
+	 */
+	if ( context.query.status === undefined ) {
+		context.query.status = DEFAULT_SITE_LAUNCH_STATUS_GROUP_VALUE;
+		return next();
+	}
+
+	const status = context.query.status.trim();
+
+	if ( status === '' ) {
+		context.page.replace( removeQueryArgs( context.canonicalPath, 'status' ) );
+		return;
+	}
+
+	const chosenStatus = getStatusFilterValue( status );
+
+	if ( ! chosenStatus ) {
+		context.page.replace( removeQueryArgs( context.canonicalPath, 'status' ) );
+		return;
+	}
+
+	context.query.status = chosenStatus;
+	next();
+}
+
+export function sitesDashboard( context: Context, next: () => void ) {
+	const sitesDashboardGlobalStyles = css`
```

Comment:
This is just duplicating v1's code (the old one will be deleted after the launch).